integreat_cms/cms/views/pages/page_form_view.py

```diff
@@ -288,6 +288,30 @@ def post(self, request, *args, **kwargs):
                 page_translation_form.instance.page.translations.filter(
                     language__in=languages
                 ).update(status=status.DRAFT)
+            # If the translation is minor change and there is no major public version
+            if (page_translation_form.instance.minor_edit is True) and (
+                not page_translation_form.instance.major_public_version
+            ):
+                # and if it is the first version, save it as major public
+                if page_translation_form.instance.version == 1:
+                    page_translation_form.instance.minor_edit = False
+                    page_translation_form.save()
+                    messages.info(
+                        request,
+                        _(
+                            'Page "{}" was saved as major version. The first version is not allowed to be a minor edit.'
+                        ).format(page_translation_form.instance.title),
+                    )
+                # or if there are only draft and/or minor public version, change them to public major version
+                else:
+                    language_tree_node = region.language_node_by_slug.get(language.slug)
+                    languages = [language] + [
+                        node.language for node in language_tree_node.get_descendants()
+                    ]
+                    page_translation_form.instance.page.translations.filter(
```

Comment:
Just a thought: we could only change only one version instead of all versions. What is better? And why?

Reply:
I did it so as the case of marking a version to draft, where all the past versions are changed to draft and thoght it avoids other edge cases.

Reply:
Alright. No further objections/questions :)

Reply:
Well, to me this also doesn't make sense. Now, if there are only draft versions and you save another version as minor draft, all versions are made public?! And the message is still "The page was successfully saved as draft", even if it is now public. I guess I will open an issue for it.

Reply:
https://github.com/digitalfabrik/integreat-cms/issues/1622 for reference